akka-persistence/src/main/scala/akka/persistence/Persistent.scala

```diff
@@ -38,6 +38,15 @@ object AtomicWrite {
 }
 
 final case class AtomicWrite(payload: immutable.Seq[PersistentRepr]) extends PersistentEnvelope with Message {
+
+  // TODO: quite costy... do we need to do this? We only internally create AWs hm...
+  require(!payload.exists(_.persistenceId != payload.head.persistenceId),
+    s"AtomicWrite must contain messages for the same persistenceId, yet different persistenceIds found: ${payload.map(_.persistenceId).toSet}")
+
+  def persistenceId = payload.head.persistenceId
+  def lowestSequenceNr = payload.map(_.sequenceNr).min // this assumes they're gapless; they should be
```

Comment:
again, assumption here that they're gapless - they are because we create them.

Well, not even that, we could `.head.sequenceNr` and `.last.sequenceNr`, if we "trust our impl, and noone else creates those".